node/overseer/overseer-gen/src/lib.rs

```diff
@@ -0,0 +1,519 @@
+//!             .                                                                .
+//!             .  +-------------------+                +---------------------+  .
+//!             .  |   Subsystem1      |                |   Subsystem2        |  .
+//!             .  +-------------------+                +---------------------+  .
+//!             .           |                                       |            .
+//!             ..................................................................
+//!                         |                                       |
+//!                       start()                                 start()
+//!                         V                                       V
+//!             ..................| Overseer "runs" these |.......................
+//!             .  +--------------------+               +---------------------+  .
+//!             .  | SubsystemInstance1 | <-- bidir --> | SubsystemInstance2  |  .
+//!             .  +--------------------+               +---------------------+  .
+//!             ..................................................................
+//! ```
+
+// #![deny(unused_results)]
+// unused dependencies can not work for test and examples at the same time
+// yielding false positives
+#![deny(missing_docs)]
+#![deny(unused_crate_dependencies)]
+
+pub use polkadot_overseer_gen_proc_macro::overlord;
+
+#[doc(hidden)]
+pub use tracing;
+#[doc(hidden)]
+pub use metered;
+#[doc(hidden)]
+pub use sp_core::traits::SpawnNamed;
+
+#[doc(hidden)]
+pub use futures::{
+	self,
+	select,
+	poll,
+	future::{
+		Fuse, Future, BoxFuture
+	},
+	stream::{
+		self, select, FuturesUnordered,
+	},
+	task::{
+		Poll, Context,
+	},
+	channel::{mpsc, oneshot},
+};
+#[doc(hidden)]
+pub use std::pin::Pin;
+#[doc(hidden)]
+pub use async_trait::async_trait;
+
+#[doc(hidden)]
+pub use std::time::Duration;
+use std::sync::atomic::{self, AtomicUsize};
+use std::sync::Arc;
+
+#[doc(hidden)]
+pub use futures_timer::Delay;
+#[doc(hidden)]
+pub use futures_util::stream::StreamExt;
+#[doc(hidden)]
+pub use futures_util::future::FutureExt;
+
+
+pub use polkadot_node_network_protocol::WrongVariant;
+
+/// A type of messages that are sent from [`Subsystem`] to [`Overseer`].
+///
+/// Used to launch jobs.
+pub enum ToOverseer {
+	/// A message that wraps something the `Subsystem` is desiring to
+	/// spawn on the overseer and a `oneshot::Sender` to signal the result
+	/// of the spawn.
+	SpawnJob {
+		/// Name of the task to spawn which be shown in jaeger and tracing logs.
+		name: &'static str,
+		/// The future to execute.
+		s: BoxFuture<'static, ()>,
+	},
+
+	/// Same as `SpawnJob` but for blocking tasks to be executed on a
+	/// dedicated thread pool.
+	SpawnBlockingJob {
+		/// Name of the task to spawn which be shown in jaeger and tracing logs.
+		name: &'static str,
+		/// The future to execute.
+		s: BoxFuture<'static, ()>,
+	},
+}
+
+use std::fmt;
```

Comment:
```suggestion
```

Reply:
I'll move it up, but it's needed two lines below.